controllers/storagecluster/cephcluster.go

```diff
@@ -266,7 +266,7 @@ func (obj *ocsCephCluster) ensureCreated(r *StorageClusterReconciler, sc *ocsv1.
 	} else {
 		// Interpret CephCluster status and set any negative conditions
 		if sc.Spec.ExternalStorage.Enable {
-			statusutil.MapExternalCephClusterNegativeConditions(&r.conditions, found)
+			statusutil.MapExternalCephClusterConditions(&r.conditions, found)
```

Comment:
If you look at other places, we only set negative conditions here. So existing implementation is fine.

If you want to set positive condition, it is done in reconcile.go at the end of reconcile if there are no negative conditions.

Also IIRC, we made a choice to not have "connected" as a condition in StorageCluster. Instead we want to keep using the existing conditions set by `SetCompleteCondition` utility function.

I see the issue that you are trying to fix, but what is the root cause? We've had this code for a while and I haven't seen any similar issue.

Reply:
Only a few conditions were set by `SetCompleteCondition`. It does not remove or update the `ExternalClusterConnecting` and `ExternalClusterConnected` conditions. These conditions are only set once. And only in a negative way. As soon as the Ceph cluster is connected, someone needs to make them positive.

Reply:
When cluster is connected, we set sc.status.phase to Ready. So after we call `SetCompleteCondition` we can check phase and remove the negative conditions. We already do that for ClusterExpanding phase.

Reply:
Do you really want to remove the conditions, I don't think it is a good practice to remove the conditions. Instead, we should change the status of the conditions.

Reply:
I don't want the conditions to change based on deployment mode. It should remain the same. I don't think there's any issue with removing negative conditions.

Reply:
I made the change as you suggested but I still don't think we should remove conditions as we can make them positive.

Reply:
We can always change it if it doesn't work out. Current change looks good to me.